### ISSUE TITLE
improvement in statistics

### ISSUE THREAD

**Issue by shanmukhananda** [paul-michalik/ORB_SLAM2#82]
@paul-michalik, @apattnaik0721013 

I see you have pushed some some change to statistics files, but I see below issues that I would like to point out.

- [Class name qualification for inline members](https://github.com/paul-michalik/ORB_SLAM2/blob/cb318d55fcd364982139e046e98ab7df4dfaef3e/ext/statistics.cc#L29), is incorrect. Not sure how this is even compiling. This giving compilation error in my ubuntu machine. 
- [Naming convention for class members](https://github.com/paul-michalik/ORB_SLAM2/blob/cb318d55fcd364982139e046e98ab7df4dfaef3e/ext/statistics.cc#L26). Follow a standard naming convention, in `statistics` class, data member begin with `_` but `stat_collector`, data member begins with `m_`
- Follow standard ordering for `public` and `private` members. `statistics` and `stat_collector` follow opposite orders. Better we choose `public` first and `private` next ordering for all classes.
- [cal_percent](https://github.com/paul-michalik/ORB_SLAM2/blob/cb318d55fcd364982139e046e98ab7df4dfaef3e/ext/statistics.h#L18) despite of its strange name, can be made private, along with few other functions like `write_row`, `write_col_list`. Use `private` always and use `public` only if its required.
- [Counters are integers?!](https://github.com/paul-michalik/ORB_SLAM2/blob/cb318d55fcd364982139e046e98ab7df4dfaef3e/ext/statistics.h#L33): Integers counters are not required in our case, as we do not deal with negative values. Better to use `std::size_t`
- [const string reference missing here?](https://github.com/paul-michalik/ORB_SLAM2/blob/cb318d55fcd364982139e046e98ab7df4dfaef3e/ext/statistics.h#L30) and [here?](https://github.com/paul-michalik/ORB_SLAM2/blob/cb318d55fcd364982139e046e98ab7df4dfaef3e/ext/statistics.cc#L50). Ignore if its expected.

**Comment by paul-michalik** [comment]
> Class name qualification for inline members, is incorrect.
You are right. I missed this completely. MSVC yells warnings at you, gcc accept this only with `-fpermissive`
> Naming convention for class members.
Agree, fix it! Whatever you choose (I prefer the underscore) be consistent! 
> Follow standard ordering for public and private members. 
I prefer `private` first (without explicit qualifier) and `public` second. But whatever you do, do not mix up `public` and `private sections.`

Other parts are also worth a fix, @apattnaik0721013 please fix this on a new branch!
